### PR TITLE
21 signup page

### LINKED ISSUE #21: Sign Up Page
**GIVEN** I am an unregistered User
**WHEN** I am on the home page
**AND** I press the Sign Up button
**THEN** I am taken to the Sign Up Page

**GIVEN** I am an unregistered User
**WHEN** I access the Sign Up Page
**AND** I fill out the form correctly
**AND** I press the Register button
**THEN** I get a success message saying “Thank you for registering”
**AND** I am logged in

**GIVEN** I am an unregistered User
**WHEN** I access the Sign Up Page
**AND** I fill out  the form with errors
**AND** I press the Register Button
**THEN** I get an error message saying “Error validating your information”
**AND** The field with the error should be highlighted

### PR DESCRIPTION
**Pull request by mattwr18**
#### Issue addressed

fixes #21 

#### Screenshots (if appropriate):

screenshot at 2019-01-15 19 19 37

#### Testing

no integration tests, but the coverage is quite high

### CODE REVIEW

**Comment by mattwr18** [comment]
getting this type of output on circleci
```
Unarchiving cache...
tar: root/WebsiteOne-FE/node_modules: Cannot mkdir: Permission denied
tar: root/WebsiteOne-FE/node_modules/.bin: Cannot mkdir: Permission denied
tar: root/WebsiteOne-FE/node_modules/.bin/JSONStream: Cannot open: Permission denied
tar: root/WebsiteOne-FE/node_modules/.bin/acorn: Cannot open: Permission denied
tar: root/WebsiteOne-FE/node_modules/.bin/ansi-html: Cannot open: Permission denied
tar: root/WebsiteOne-FE/node_modules/.bin/atob: Cannot open: Permission denied
```
it's passing on travis though

File: src/containers/SignUp.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  handleSignUp = async e => {
+    const { email, password } = this.state
+    e.preventDefault()
+    await this.props
+      .postSignUpInfo({ email, password })
+      .then(() => {
+        iziToast.show({
+          theme: 'light',
+          title: 'Success',
+          message: 'Have a look around',
```
Comment (joaopapereira) [inline src/containers/SignUp.js]: Message is not same as in story

File: src/tests/containers/SignUp.test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  const emailInput = wrapper.find('input').filterWhere(item => {
+    return item.prop('name') === 'email'
+  })
+
+  const passwordInput = wrapper.find('input').filterWhere(item => {
+    return item.prop('name') === 'password'
+  })
+
+  it('should have a SignUp Header', () => {
+    const mainHeader = wrapper.find('Header').filterWhere(item => {
```
Comment (joaopapereira) [inline src/tests/containers/SignUp.test.js]: Not sure about the value of the test

File: src/tests/containers/SignUp.test.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      return item.text() === 'Sign Up'
+    })
+    expect(mainHeader).toHaveLength(1)
+  })
+
+  it('should have a Form', () => {
```
Comment (joaopapereira) [inline src/tests/containers/SignUp.test.js]: Not sure about the value of the test

File: src/tests/containers/SignUp.test.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    expect(wrapper.find('Form')).toHaveLength(1)
+  })
+
+  it('should update the state when email input value changes', () => {
```
Comment (joaopapereira) [inline src/tests/containers/SignUp.test.js]: Not sure about the value of the test

File: src/tests/containers/SignUp.test.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    emailInput.simulate('change', {
+      target: { value: 'existing-user@example.com' }
+    })
+    expect(wrapper.state().email).toBe('existing-user@example.com')
+  })
+
+  it('should update the state when password input value changes', () => {
```
Comment (joaopapereira) [inline src/tests/containers/SignUp.test.js]: Not sure about the value of the test

File: src/tests/containers/SignUp.test.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    passwordInput.simulate('change', {
+      target: { value: 'password' }
+    })
+    expect(wrapper.state().password).toBe('password')
+  })
+
+  it('should call handleSignUp and postSignUpInfo when the form is submitted', async () => {
```
Comment (joaopapereira) [inline src/tests/containers/SignUp.test.js]: Missing the failure case

File: src/tests/reducers/usersReducer.test.js [hunk 7]
```diff
@@ -14,4 +14,10 @@ describe('reduces users', () => {
       })
     ).toEqual(['User to be added to store'])
   })
+  expect(
```
Comment (joaopapereira) [inline src/tests/reducers/usersReducer.test.js]: Maybe it would be it a separate it block

**Review by joaopapereira — CHANGES_REQUESTED** [review]
Good job on sign up. I do have some questions and comment, please review them

**Review by joaopapereira — APPROVED** [review]
Looks good, going to merge

**Pull request merged**